gym/envs/box2d/car_racing.py

```diff
@@ -438,235 +436,209 @@ def step(self, action):
+        trans = (WINDOW_W / 2 + trans[0], WINDOW_H / 4 + trans[1])
 
-        self.car.draw(self.viewer, mode != "state_pixels")
+        self.render_road(zoom, trans, angle)
+        self.car.draw(self.surf, zoom, trans, angle, mode != "state_pixels")
 
-        arr = None
-        win = self.viewer.window
-        win.switch_to()
-        win.dispatch_events()
+        self.surf = pygame.transform.flip(self.surf, False, True)
 
-        win.clear()
-        t = self.transform
-        if mode == "rgb_array":
-            VP_W = VIDEO_W
-            VP_H = VIDEO_H
-        elif mode == "state_pixels":
-            VP_W = STATE_W
-            VP_H = STATE_H
-        else:
-            pixel_scale = 1
-            if hasattr(win.context, "_nscontext"):
-                pixel_scale = (
-                    win.context._nscontext.view().backingScaleFactor()
-                )  # pylint: disable=protected-access
-            VP_W = int(pixel_scale * WINDOW_W)
-            VP_H = int(pixel_scale * WINDOW_H)
-
-        gl.glViewport(0, 0, VP_W, VP_H)
-        t.enable()
-        self.render_road()
-        for geom in self.viewer.onetime_geoms:
-            geom.render()
-        self.viewer.onetime_geoms = []
-        t.disable()
+        self.screen.fill(0)
+        self.screen.blit(self.surf, (0, 0))
+
+        # showing stats
         self.render_indicators(WINDOW_W, WINDOW_H)
 
-        if mode == "human":
-            win.flip()
-            return self.viewer.isopen
+        font = pygame.font.Font(pygame.font.get_default_font(), 42)
+        text = font.render("%04i" % self.reward, True, (255, 255, 255), (0, 0, 0))
+        text_rect = text.get_rect()
+        text_rect.center = (60, WINDOW_H - WINDOW_H * 2.5 / 40.0)
+        self.screen.blit(text, text_rect)
 
-        image_data = (
-            pyglet.image.get_buffer_manager().get_color_buffer().get_image_data()
-        )
-        arr = np.fromstring(image_data.get_data(), dtype=np.uint8, sep="")
-        arr = arr.reshape(VP_H, VP_W, 4)
-        arr = arr[::-1, :, 0:3]
-
-        return arr
+        if mode == "human":
+            pygame.display.flip()
 
-    def close(self):
-        if self.viewer is not None:
-            self.viewer.close()
-            self.viewer = None
-
-    def render_road(self):
-        colors = [0.4, 0.8, 0.4, 1.0] * 4
-        polygons_ = [
-            +PLAYFIELD,
-            +PLAYFIELD,
-            0,
-            +PLAYFIELD,
-            -PLAYFIELD,
-            0,
-            -PLAYFIELD,
-            -PLAYFIELD,
-            0,
-            -PLAYFIELD,
-            +PLAYFIELD,
-            0,
+        if mode == "rgb_array":
+            scaled_screen = pygame.transform.smoothscale(
+                self.screen, (VIDEO_W, VIDEO_H)
+            )
+            return np.transpose(
+                np.array(pygame.surfarray.pixels3d(scaled_screen)), axes=(1, 0, 2)
+            )
```

Comment:
Could be worth it to offload this to a separate method:

```python
def _create_image_array(screen):
    return np.transpose(
                np.array(pygame.surfarray.pixels3d(screen)), axes=(1, 0, 2)
            )
```
and use it here in the two cases.

Reply:
I added the an additional parameter which takes in the dimensions needed and scaled the screen accordingly. It's been refactored